### PR TITLE
Products

### PR DESCRIPTION
**Pull request by bossmonkey98**
Filter functionality addded

### CODE REVIEW

File: src/components/context/filter-context.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { createContext,useContext,useReducer } from "react";
+import FilterReducer from '../filter'
```
Comment (t007rushi) [inline src/components/context/filter-context.js:2]: nice that you take out reducer out

File: src/components/context/product-context.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import axios from 'axios';
+import { useState } from "react";
+
+const ProductContext = createContext()
+
+function ProductProvider({children}){
+    const [ProductData, setProductData] =useState();
+    const url ="/api/products"
+        const fetchProductdata = async() =>{
+            const response = await axios.get(url)
```
Comment (t007rushi) [inline src/components/context/product-context.js:11]: you can destructure here {data}

File: src/components/filter.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    return product.sort((a, b) => a.price - b.price);
+}
+    else return product;
+}
+
+function getratingdata (product,rating){
+    return product.filter((items)=>items.rating>=rating)
+}
+
+export function filtereddata(productList,{sortBy,showAccessories,showApparel,showCollectibles,rating}){
```
Comment (t007rushi) [inline src/components/filter.jsx:46]: can make them as service

File: src/components/categories.jsx [hunk 4]
```diff
@@ -13,8 +13,12 @@ const Wrapper = styled.div`
...
-margin:4px;
-object-fit:contain;`
+margin:1rem;
+object-fit:contain;
+&:hover{
+    transform: scale(1.1);
+    transition: 0.8s ease;
+}`;
 
 const Image = styled.img`
```
Comment (sanjitsarkar) [inline src/components/categories.jsx:23]: Is it okay to use styled component?

File: src/components/categories.jsx [hunk 5]
```diff
@@ -47,9 +54,7 @@ color: white;
 border-radius: 5px;
 background-color:red;
 &:hover{
-    font-weight: 600;
    background-color:var(--bg-color);
-   color:red;
    border:1px solid #c30101;
```
Comment (sanjitsarkar) [inline src/components/categories.jsx:58]: use variable for color

**Review by sanjitsarkar — APPROVED** [review]

File: src/pages/Productlisting/ProductListing.jsx [hunk 6]
```diff
@@ -1,182 +1,56 @@
...
 function ProductListing() {
-  return (
+    const {ProductData, fetchProductdata} =useProducts()
+    useEffect(()=>fetchProductdata(),[])
+    const {state ,dispatch} =useFilter()
+    const {sortBy,showAccessories,showApparel,showCollectibles,rating} = {...state}
+    const prod = ProductData
+    var products = filtereddata(prod,state)
+    console.log(state)
+    console.log(products)
```
Comment (t007rushi) [inline src/pages/Productlisting/ProductListing.jsx]: remove consoles

**Review by t007rushi — APPROVED** [review]
Look for suggestions

**Pull request merged**